src/api/journalEntries/journalEntry.ts

```diff
@@ -158,6 +168,34 @@ export const updateJournalEntry = async (updateJournalEntryInput: JournalEntryUp
     mutation: gql(updateJournalEntryQuery),
     variables: { data: variables, where: { id: Number(id) }}
   });
-  
+
   return data.updateJournalEntry;
+}
+
+const deleteJournalEntryMutation = gql`
+  mutation deleteJournalEntry($where: JournalEntryWhereUniqueInput!) {
+    deleteJournalEntry(where: $where) {
+      id
+    }
+  }
+`
+
+const deleteJournalEntry = async (id: number) => {
+  const { data } = await API.mutate<any>({
+    mutation: deleteJournalEntryMutation,
+    variables: { where: { id: Number(id) } }
+  });
+
+  return data.deleteJournalEntry;
+}
+
+export const useDeleteJournalEntry = () => {
+  const queryClient = useQueryClient();
+
+  return useMutation(deleteJournalEntry, {
+    onSuccess: (_, id) => {
```

Comment:
The onSuccess will receive whatever data is returned from the mutation function, in this case delete will only give back the id property and the typename. The cached data is filtered to exclude the id of the successfully deleted entry. This saves having to make another query but is still valid because its only triggered onSuccess. There will always be cached data because the user can only delete from the dashboard at the moment, so invalidating the cache is unnecessary